### DISCUSSION TITLE
sylpheed: cannot apply dark theme when using --private=

### DISCUSSION THREAD

**Discussion by Rosika2** [netblue30/firejail#5397] [Q&A]
Hi all,

Info:
my system: Linux/Lubuntu 20.04.5 LTS, 64bit

a while ago I discussed an issue ( problems with sylpheed #4995 ) with @glitsj16 ).
It was about running the e-mail client sylpheed with firejail using a **dedicated work-directory**.
That problem could be solved. I´m using the command
`	 firejail --private-etc=machine-id  --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed sylpheed`
now and everything works perfectly.

There´s just another question however that popped into my mind:

On my system there is **dark theme** enabled. I.e. GUI programmes normally use that. I configured it in **system settings** thus:

- 	LXQt Appearance Configuration:
- 	Set GTK themes (GTK configuration files will be overwritten): checked
- 	GTK2 theme: Arc Dark
- 	GTK3 theme: Arc Dark
- 	LXQt theme: Dark

	
The thing however is: when I open sylpheed with the above-mentioned command I don´t get the dark theme. Everything is displayed black-on-white.
BUT: if I open sylpheed this way:
`	firejail sylpheed`
the dark theme setting IS applied (white-on-black).

... i.e. it´s just when running sylpheed in firejail in a dedicated work-directory that I cannot get dark theme.

Any ideas how it could be done?

Thanks very much in advance.
Many greetings
Rosika  :-)

**Comment by rusty-snake** [comment]
AFAIK
- sylpheed uses GTK+ 2.0
- GTK+ 2.0 theme is configured only in `~/.gtkrc-2.0` (and `/etc`)

=> Try to copy `cp ~/.gtkrc-2.0 /media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed`

**Reply by rusty-snake** [reply]
There's also a `GTK2_RC_FILES` env-var, see https://wiki.archlinux.org/title/GTK#Themes.

**Comment by ghost** [comment]
To make that work I assume you would need to have the relevant GTK theme files in your dedicated work-dir. Sylpheed is GTK2 only I reckon, so you'd only need to copy the GTK2 Arc Dark theme to a .themes subdir inside your dedicated location. Then you should be able to start sylpheed with an extra environment variable to force it to use that theme:
```
$ GTK2_RC_FILES=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed/.themes/ArcDarc/gtk-2.0/gtkrc \
firejail --private-etc=machine-id --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed sylpheed
```

Due to the length of these commands it would be easier to create a simple alias in your shell or write a wrapper script.

**Comment by Rosika2** [comment]
@rusty-snake and @glitsj16:

Thanks a lot to both of you for your answers and your great help.

What you both suggested is the solution indeed. 
So basically I got two versions which both work equally well:

BTW: I use `fish` as my default shell; so setting an evironment variable for a command is done with `env` prepended.

1.) ` env GTK2_RC_FILES=/usr/share/themes/Arc-Dark/gtk-2.0/gtkrc firejail --private-etc=machine-id  --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed sylpheed`

2.)` env GTK2_RC_FILES=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed/.themes/Arc-Dark/gtk-2.0/gtkrc firejail --private-etc=machine-id --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed sylpheed`

For 2.) to work I copied 
`cp -r /usr/share/themes/Arc-Dark /media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/kgw_work3_sylpheed/.themes/`
into the newly created folder `.themes` in the work directory like @glitsj16 suggested.

Thanks for providing the solution. It´s highly appreciated.
I´ve also learned quite a bit again this way.

BTW:

> Due to the length of these commands it would be easier to create a simple alias in your shell or write a wrapper script.

Thanks for the suggestion. It´s a great idea of course.
Yet, as I use the `fish`-shell it´s easy to use the most complex commands in a very simple way:
I just have to type only a fraction of the command like "syl", then hit "Arrow up" and there´s the command [like in 1.) or 2.)]. I just hit "enter" then  and  I´m done.
That´s of course provided I haven´t used another command in the meantime which has "syl" as a part of it.
In this unlikely case I can type "sylpheed" then...

Thanks again and many greetings from Rosika.

P.S.:

Both of your replies are splendid. I really don´t know which to mark as "solution" without offending anyone of you.
I seem to be in a pickle here...

P.S.2:

Thanks @rusty-snake for providing the arch wiki link.
